src/utilities/oas-validator.ts

```diff
@@ -243,6 +259,72 @@ class OASValidator {
     return failures;
   }
 
+  private checkMissingRequiredParameters(
+    operation: Method | null,
+    parameters: ParameterExamples,
+  ): ValidationFailure | undefined {
+    const requiredParameters = operation?.parameters
+      .filter((parameter) => parameter.required)
+      .map((parameter) => parameter.name);
+
+    const presentParameterNames = Object.keys(parameters);
+    const missingRequiredParameters = requiredParameters?.filter(
+      (parameterName) => !presentParameterNames.includes(parameterName),
+    );
+
+    if (missingRequiredParameters && missingRequiredParameters.length > 0) {
+      return new MissingRequiredParameters(missingRequiredParameters);
+    }
+  }
+
+  private checkParameterObject(
+    parameterObject,
+    path: string[],
+  ): CheckParameterObject {
+    const checkObject: CheckParameterObject = {
+      schema: null,
+      parameterObjectFailure: null,
+    };
+    if (parameterObject.schema && !parameterObject.content) {
+      checkObject.schema = parameterObject.schema;
+      return checkObject;
+    }
+
+    if (parameterObject.content) {
+      if (parameterObject.schema) {
+        checkObject.parameterObjectFailure = new InvalidParameterObject(path);
+        return checkObject;
+      }
+
+      const [contentObjectKey, ...invalidKeys] = Object.keys(
+        parameterObject.content,
+      );
+      if (invalidKeys.length > 0) {
+        checkObject.parameterObjectFailure = new InvalidParameterContent([
+          ...path,
+          'content',
+        ]);
+
+        return checkObject;
+      }
+
+      if (parameterObject.content[contentObjectKey].schema) {
+        checkObject.schema = parameterObject.content[contentObjectKey].schema;
+        return checkObject;
+      }
+
+      checkObject.parameterObjectFailure = new MissingSchemaObject([
+        ...path,
+        'content',
+        contentObjectKey,
+      ]);
+      return checkObject;
+    }
+
+    checkObject.parameterObjectFailure = new InvalidParameterObject(path);
```

Comment:
I think this failure is for when there is neither a schema nor a content? It's not super clear though, may be a good thing to refactor this method to make this more explicit

Reply:
I've added comments to make it clearer what's happening at each "gate" let me know what you think!